// @insforge/shared-schemas/dist/deno-deploy.schema.js
import { z } from 'zod';
/**
 * Deno Deploy deployment status
 */
export const denoDeployStatusSchema = z.enum(['pending', 'deploying', 'success', 'failed']);
/**
 * Function definition for deployment
 */
export const denoFunctionDefinitionSchema = z.object({
    slug: z.string(),
    code: z.string(),
});
/**
 * Deno Deploy deployment result
 */
export const denoDeployResultSchema = z.object({
    id: z.string(),
    projectId: z.string(),
    status: denoDeployStatusSchema,
    url: z.string().nullable(),
    domains: z.array(z.string()),
    createdAt: z.date(),
    updatedAt: z.date(),
    error: z
        .object({
        code: z.string(),
        message: z.string(),
    })
        .optional(),
});
/**
 * Deno Deploy asset file
 */
export const denoDeployAssetSchema = z.object({
    kind: z.literal('file'),
    content: z.string(),
    encoding: z.literal('utf-8'),
});
/**
 * Deno Deploy API response
 */
export const denoDeployApiResponseSchema = z.object({
    id: z.string(),
    projectId: z.string(),
    description: z.string().nullable(),
    status: z.enum(['pending', 'success', 'failed']),
    domains: z.array(z.string()),
    databases: z.record(z.string()),
    requestTimeout: z.number().nullable(),
    permissions: z.record(z.unknown()).nullable(),
    createdAt: z.string(),
    updatedAt: z.string(),
});
